src/JustSaying/AwsTools/MessageHandling/Dispatch/MessageDispatcher.cs

```diff
@@ -78,6 +82,15 @@ await middleware.RunAsync(handleContext, null, cancellationToken)
         {
             _logger.LogDebug("Attempting to deserialize message with serialization register {Type}",
                 _serializationRegister.GetType().FullName);
+
+            messageContext.Message.MessageAttributes.TryGetValue("Content-Encoding", out var contentEncoding);
+            if (contentEncoding is not null)
+            {
+                var decompressor = _decompressionRegistry.GetDecompressor(contentEncoding.StringValue);
+                // TODO What to do when decompressor not found?
```

Comment:
This created some interesting discussion, as my instinct is to throw here, if we see there is a `Content-Encoding` mentioning a compression we weren't expecting, however @hwoodiwiss pointed out that this could be a header someone is already legitimately using for other purposes, and we just broke them.

We could have some sort of opt-in behaviour, and if the default "no compression" configuration is detected then skip this block entirely. I was thinking we just had it on for subscription by default, but we might want to make that a major or minor version bump.